src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ApplyChangesToWorkspaceContext.cs

```diff
@@ -150,8 +150,15 @@ private void ProcessOptions(IProjectRuleSnapshot snapshot)
         {
             Assumes.NotNull(_context);
 
+            // The order of options matters and we should use warnaserror- and then warnaserror+
+            // Depending which comes first, the compiler produces different diagnostics.
+            // We just need to move warnaserror+ to the end.
+            var warnaserrorPlus = snapshot.Items.Where(a => a.Key.Contains("/warnaserror+")).FirstOrDefault();
+            ImmutableArray<string> commandLineArguments = warnaserrorPlus.Key is null ? snapshot.Items.Keys.ToImmutableArray() : 
+                snapshot.Items.Remove(warnaserrorPlus.Key).Keys.ToArray().Append(warnaserrorPlus.Key).ToImmutableArray();
+ 
```

Comment:
This is very inefficient and will start pushing lots of data on the LOH heap, increasing GC time. This has been a large problem in the past, and was written to be much more efficient. If this approach is taken, please rewrite to avoid inbetween allocations (ToArray()) allocations